flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/functions/utils/UserDefinedFunctionUtils.scala

```diff
@@ -749,6 +749,9 @@ object UserDefinedFunctionUtils {
     val paraInternalType = fromDataTypeToLogicalType(parameterType)
     if (isAny(internal) && isAny(paraInternalType)) {
       getDefaultExternalClassForType(internal) == getDefaultExternalClassForType(paraInternalType)
+    } else if ((isTimestamp(internal) && isLong(paraInternalType))
```

Comment:
not sure about this. 
cc @JingsongLi

Reply:
ditto, logical type already defined the conversions.